sensing/position_estimation.cpp

```diff
@@ -356,41 +336,47 @@ Position_estimation::Position_estimation(State& state, Barometer& barometer, con
         sonar(sonar)
 {
     // default GPS home position
-    local_position.origin.longitude =  config.origin.longitude;
-    local_position.origin.latitude =   config.origin.latitude;
-    local_position.origin.altitude =   config.origin.altitude;
+    origin_.longitude =  config.origin.longitude;
+    origin_.latitude =   config.origin.latitude;
+    origin_.altitude =   config.origin.altitude;
 
     for(uint8_t i = 0; i < 3; i++)
     {
-        local_position.pos[i] = 0.0f;
+        local_position[i] = 0.0f;
         kp_pos_gps[i] = config.kp_pos_gps[i];
         kp_vel_gps[i] = config.kp_vel_gps[i];
     }
-
-    if (fence_set)
-    {
-        set_new_fence_origin();
-    }
 }
 
 
-void Position_estimation::update()
+bool Position_estimation::update(void)
 {
+    // Updat timing
+    float now      = time_keeper_get_s();
+    dt_s_          = now - last_update_s_;
+    last_update_s_ = now;
+
     if (ahrs.internal_state == AHRS_READY)
     {
         if (state.reset_position)
         {
             state.reset_position = false;
             reset_home_position();
-            set_new_fence_origin();
```

Comment:
local_position_t does not have an origin field, so the method is not needed anymore.